exchanges/okcoin/okcoin_websocket.go

```diff
@@ -246,43 +532,50 @@ func (o *OKCoin) wsProcessOrder(respRaw []byte) error {
 	return nil
 }
 
-// wsProcessTickers converts ticker data and sends it to the datahandler
-func (o *OKCoin) wsProcessTickers(respRaw []byte) error {
-	var response WebsocketTickerData
-	err := json.Unmarshal(respRaw, &response)
+// wsProcessInstruments converts instrument data and sends it to the datahandler
+func (o *OKCoin) wsProcessInstruments(respRaw []byte) error {
+	var response []WebsocketInstrumentData
+	resp := WebsocketDataResponseReciever{
+		Data: &response,
+	}
+	err := json.Unmarshal(respRaw, &resp)
 	if err != nil {
 		return err
 	}
-	a := o.GetAssetTypeFromTableName(response.Table)
-	for i := range response.Data {
-		f := strings.Split(response.Data[i].InstrumentID, delimiterDash)
-
-		c := currency.NewPairWithDelimiter(f[0], f[1], delimiterDash)
-
-		baseVolume := response.Data[i].BaseVolume24h
-		if response.Data[i].ContractVolume24h != 0 {
-			baseVolume = response.Data[i].ContractVolume24h
-		}
+	o.Websocket.DataHandler <- response
+	return nil
+}
 
-		quoteVolume := response.Data[i].QuoteVolume24h
-		if response.Data[i].TokenVolume24h != 0 {
-			quoteVolume = response.Data[i].TokenVolume24h
+// wsProcessTickers  converts ticker data and sends it to the datahandler
+func (o *OKCoin) wsProcessTickers(respRaw []byte) error {
+	var response []WsTickerData
+	resp := WebsocketDataResponseReciever{
+		Data: &response,
+	}
+	err := json.Unmarshal(respRaw, &resp)
+	if err != nil {
+		return err
+	}
+	for x := range response {
+		pair, err := currency.NewPairFromString(response[x].InstrumentID)
+		if err != nil {
+			return err
 		}
-
 		o.Websocket.DataHandler <- &ticker.Price{
```

Comment:
suggestion: send a slice instead

Reply:
```
case []ticker.Price:
		for x := range d {
			if m.syncer.IsRunning() {
				err := m.syncer.Update(exchName,
					d[x].Pair,
					d[x].AssetType,
					SyncItemTicker,
					nil)
				if err != nil {
					return err
				}
			}
			err := ticker.ProcessTicker(&d[x])
			if err != nil {
				return err
			}
			m.syncer.PrintTickerSummary(&d[x], "websocket", err)
		}
```
If so, I also add this to **_websocketroutine_manager.go_**

Reply:
Yes please